### PR TITLE
DOCSP-38412 - Connect Usage Examples

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-38412
Staging - https://preview-mongodbmongokart.gatsbyjs.io/pymongo/docsp-38412-connect-usage/connect/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/connect.txt [hunk 1]
```diff
@@ -4,8 +4,19 @@
...
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
 .. meta::
-   :description: Learn how to use (+driver-short+} to connect to from MongoDB.
+   :description: Learn how to use (+driver-short+} to connect to MongoDB.
```
Comment (jordan-smith721) [inline source/connect.txt:18]:
```suggestion
   :description: Learn how to use {+driver-short+} to connect to MongoDB.
```

File: source/connect.txt [hunk 2]
```diff
@@ -17,8 +28,199 @@ Connect to MongoDB
...
+To use a connection example from this page, copy the code example into the
+:ref:`sample application <pymongo-connect-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<hostname>``, with
+the relevant values for your MongoDB deployment.
+
+.. _pymongo-connect-sample:
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/connect-sample-app.py
```
Comment (jordan-smith721) [inline source/connect.txt:51]: Can this use the same sample app as authentication?

File: source/connect.txt [hunk 3]
```diff
@@ -17,8 +28,199 @@ Connect to MongoDB
...
+   client = pymongo.MongoClient("mongodb://<username>:<password>@<hostname>:<port>",
+                                     server_selector=<selector function>)
+
+To learn more about customizing server selection, see
+:ref:`pymongo-server-selection`.
+
+{+stable-api+}
+--------------
+
+.. literalinclude:: /includes/connect/stable_api.py
```
Comment (jordan-smith721) [inline source/connect.txt:184]: Is this is intentionally blank for code generation?

File: source/security.txt [hunk 4]
```diff
@@ -49,8 +49,6 @@ the relevant values for your MongoDB deployment.
 .. literalinclude:: /includes/usage-examples/auth-sample-app.py
       :language: python
       :copyable: true
-      :start-after: # start sample app
-      :end-before: # end sample app
       :linenos:
       :emphasize-lines: 4-6
```
Comment (jordan-smith721) [inline source/security.txt:53]: Doesn't look like it's breaking anything but the indentation is off here

**Review by jordan-smith721 — APPROVED** [review]
LGTM w/ a couple small things

File: source/connect.txt (hunk 1 above)
Comment (mongoKart) [reply source/connect.txt:18]: someday i will stop doing this, maybe

File: source/connect.txt (hunk 2 above)
Comment (mongoKart) [reply source/connect.txt:51]: sure--i just made the auth sample a little more generic to accommodate

File: source/connect.txt (hunk 3 above)
Comment (mongoKart) [reply source/connect.txt:184]: It is, but I might as well add it now

File: source/connect.txt [hunk 5]
```diff
@@ -17,8 +28,201 @@ Connect to MongoDB
...
+
+Limit Server Execution Time
+---------------------------
+
+timeout Block
+~~~~~~~~~~~~~
+
+.. code-block:: python
+
+   with pymongo.timeout(<timeout length>):
```
Comment (blink1073) [inline source/connect.txt:202]: I'd recommend adding a comment below, so it looks something like:

```python
with pymongo.timeout(<timeout length>): 
      # perform operations here
```

**Pull request merged**